CMakeLists.txt

```diff
@@ -58,119 +60,162 @@ message(STATUS "This is a development version of Tasmanian")
 option(Tasmanian_ENABLE_FORTRAN "Enable Fortran interface for Tasmanian" OFF)
-option(Tasmanian_ENABLE_CUBLAS "Enable Nvidia GPU acceleration using cuBLAS for Tasmanian" ON)
-option(Tasmanian_ENABLE_CUDA "Enable Nvidia GPU acceleration using CUDA kernels for Tasmanian" OFF)
+option(Tasmanian_ENABLE_CUBLAS  "Enable Nvidia GPU acceleration using cuBLAS for Tasmanian"       ON)
+option(Tasmanian_ENABLE_CUDA    "Enable Nvidia GPU acceleration using CUDA kernels for Tasmanian" OFF)
 option(Tasmanian_STATIC_LIBRARY "Compile static library and the executables with static linkage (Tasmanian)" ON)
 option(Tasmanian_SHARED_LIBRARY "Compile dynamic library (Tasmanian)" ON)
 
-set(Tasmanian_TESTS_GPU_ID "-1" CACHE STRING "specify GPU ID for testing, -1 means running tests on all visible devices")
+# threat those similar to options, but give values other than ON/OFF
+set(Tasmanian_TESTS_GPU_ID          "-1" CACHE STRING "specify GPU ID for testing, -1 means running tests on all visible devices")
 set(Tasmanian_TESTS_OMP_NUM_THREADS "-1" CACHE STRING "specify OMP_NUM_THREADS for the tests, if less than 0, this option will be ignored")
-set(Tasmanian_MATLAB_WORK_FOLDER "" CACHE PATH "specify the path for the MATLAB work folder")
-
-# version and license
-set(Tasmanian_version_string "${Tasmanian_VERSION_MAJOR}.${Tasmanian_VERSION_MINOR}")
-set(Tasmanian_license "BSD 3-Clause with UT-Battelle disclaimer")
+set(Tasmanian_MATLAB_WORK_FOLDER     ""  CACHE PATH   "specify the path for the MATLAB work folder")
 
 
 ########################################################################
-# Defines the source files
-# NOTE: sanity check modifies the CUDA source lists before setup target
+# Sanity check, xSDK compatibility, and find_package() calls are all
+# done in one place for consistency
+# after this call, every enabled option comes with proper found set of
+# find_package() provided libraries, includes, etc.
 ########################################################################
-include(${CMAKE_SOURCE_DIR}/Config/CMakeIncludes/source_file_list.cmake)
+include("${CMAKE_CURRENT_SOURCE_DIR}/Config/CMakeIncludes/sanity_check_and_xsdk.cmake")
 
 
 ########################################################################
-# Sanity check, xSDK compatibility, and find_package() calls
+# Core project configuration
 ########################################################################
-include(${CMAKE_SOURCE_DIR}/Config/CMakeIncludes/sanity_check_and_xsdk.cmake)
+enable_testing() # call in one place, each subfolder gives additional tests
 
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/Config/TasmanianConfig.in.hpp"
+               "${CMAKE_CURRENT_BINARY_DIR}/configured/TasmanianConfig.hpp")
+if(${CMAKE_VERSION} VERSION_LESS "3.9.0") # check the correct version here
+    # older versions of cmake do not add incude directories for cuda targets
+    include_directories("${CMAKE_CURRENT_BINARY_DIR}/configured/")
+endif() 
```

Comment:
`target_include_directories(<configured>)` is called for all appropriate targets, but cmake prior to 3.9 doesn't pass those to `nvcc`. We require cmake `3.5` to avoid being aggressive, so older cmake files require an extra include directive to ensure those get passed to nvcc.